From: SeongJae Park <sj@kernel.org>
Date: Sun, 30 Nov 2025 13:25:11 -0800
Subject: [PATCH] mm/damon/sysfs: do not generate ops_attrs directory

Now sample control can replace it.  Remove the old one.

Signed-off-by: SeongJae Park <sj@kernel.org>
---
 mm/damon/sysfs.c | 10 +---------
 1 file changed, 1 insertion(+), 9 deletions(-)

diff --git a/mm/damon/sysfs.c b/mm/damon/sysfs.c
index 8babcae5cfb2..751d74a42432 100644
--- a/mm/damon/sysfs.c
+++ b/mm/damon/sysfs.c
@@ -1682,13 +1682,9 @@ static int damon_sysfs_context_add_dirs(struct damon_sysfs_context *context)
 {
 	int err;
 
-	err = damon_sysfs_context_set_ops_attrs(context);
-	if  (err)
-		return err;
-
 	err = damon_sysfs_context_set_attrs(context);
 	if (err)
-		goto put_ops_attrs_out;
+		return err;
 
 	err = damon_sysfs_context_set_targets(context);
 	if (err)
@@ -1705,15 +1701,11 @@ static int damon_sysfs_context_add_dirs(struct damon_sysfs_context *context)
 put_attrs_out:
 	kobject_put(&context->attrs->kobj);
 	context->attrs = NULL;
-put_ops_attrs_out:
-	kobject_put(&context->ops_attrs->kobj);
-	context->ops_attrs = NULL;
 	return err;
 }
 
 static void damon_sysfs_context_rm_dirs(struct damon_sysfs_context *context)
 {
-	kobject_put(&context->ops_attrs->kobj);
 	damon_sysfs_attrs_rm_dirs(context->attrs);
 	kobject_put(&context->attrs->kobj);
 	damon_sysfs_targets_rm_dirs(context->targets);
-- 
2.47.3

